### PR TITLE
Support false triggerBeforeAttach/triggerAttach show() options

### LINKED ISSUE #2484: Support "false" value for Region#show() options triggerBeforeAttach/triggerAttach
The Region gives the developer power to use `show()` option `{triggerBeforeAttach: true}` but not `{triggerBeforeAttach: false}`, and similarly with `triggerAttach`.

While the ability to use `show()` options set to `false` is not documented one way or another, Backbone `Collection#set()` methods establish a precedent of [disabling a default behavior with `false` method options](http://backbonejs.org/#Collection-set).

In this example, onBeforeAttach will fire, [as documented](http://marionettejs.com/docs/marionette.region.html#onbeforeattach-onattach):

```js
var FooRegion = Marionette.Region.extend({
  triggerBeforeAttach: false
});
region.show(view, {triggerBeforeAttach: true});
```

In this example, onBeforeAttach will _also_ fire (no documentation for this case):

```js
var FooRegion = Marionette.Region.extend({
  triggerBeforeAttach: true  // The default value, but presented here for clarity
});
region.show(view, {triggerBeforeAttach: false});
```

The second example does not match my expectations, partly set up by Backbone, because the explicit `false` option is ignored.

The fix for this is small (two lines).  The question is whether we want to support explicit false options, particularly in this case.

### PR DESCRIPTION
**Pull request by ianmstew**
Fixes #2484.

### CODE REVIEW

File: src/region.js [hunk 1]
```diff
@@ -112,10 +112,12 @@ Marionette.Region = Marionette.Object.extend({
       // as it's a potentially-slow method
       var displayedViews = [];
 
-      var triggerBeforeAttach = showOptions.triggerBeforeAttach || this.triggerBeforeAttach;
-      var triggerAttach = showOptions.triggerAttach || this.triggerAttach;
+      var attachOptions = _.extend({
+        triggerBeforeAttach: this.triggerBeforeAttach,
+        triggerAttach: this.triggerAttach
+      }, showOptions);
```
Comment (ahumphreys87) [inline src/region.js]: neat!

**Comment by ahumphreys87** [comment]
:+1:

File: src/region.js (hunk 1 above)
Comment (ianmstew) [inline src/region.js]: thanks!  first I tried checking explicitly for `showOptions.triggerAttach === false` but it became a pretty ugly logic statement.

Comment (jasonLaster) [inline src/region.js]: this will do. nice!

Comment (ianmstew) [inline src/region.js]: haha, i can see that since I'm pulling `this` into the extend target.

**Pull request merged**